vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -0,0 +1,293 @@
+// repositories and can be misleading as to the software's true repository,
+// Currently not used for this purpose due to undesired false positives
+// reducing the number of valid records successfully converted.
+var RefTagDenyList = []string{
+	// "Exploit",
+	// "Third Party Advisory",
+	"Broken Link", // Actively ignore these though.
+}
+
+// addIDToNotes prepends a CVE ID to a slice of notes.
+func addIDToNotes(id cves.CVEID, notes []string) []string {
+	var updatedNotes []string
+	for _, note := range notes {
+		updatedNotes = append(updatedNotes, fmt.Sprintf("[%s] %s", id, note))
+	}
+	return updatedNotes
+}
+
+// determineHeuristics examines a CVE and extracts metrics and heuristics about it.
+func determineHeuristics(cve cves.CVE5, v *vulns.Vulnerability) {
+	// CNA based heuristics
+	Metrics.CNA = cve.Metadata.AssignerShortName
+	// TODO(jesslowe): more CNA based analysis
+	// Reference based heuristics
+	// Count number of references of each type
+	refTypeCounts := make(map[osvschema.ReferenceType]int)
+	for _, ref := range v.References {
+		refTypeCounts[ref.Type]++
+	}
+	Metrics.RefTypesCount = refTypeCounts
+	for refType, count := range refTypeCounts {
+		Metrics.Notes = append(Metrics.Notes, fmt.Sprintf("[%s]: Reference Type %s: %d", cve.Metadata.CVEID, refType, count))
+	}
+}
+
+// FromCVE5 creates a Vulnerability from a CVE5 object.
+func FromCVE5(cve cves.CVE5, refs []cves.Reference) (*vulns.Vulnerability, []string) {
+	aliases, related := vulns.ExtractReferencedVulns(cve.Metadata.CVEID, cve.Metadata.CVEID, refs)
+	var err error
+	var notes []string
+	v := vulns.Vulnerability{}
+	v.SchemaVersion = osvschema.SchemaVersion
+	v.ID = string(cve.Metadata.CVEID)
+	v.Summary = string(cve.Containers.CNA.Title)
+	v.Details = cves.EnglishDescription(cve.Containers.CNA.Descriptions)
+	v.Aliases = aliases
+	v.Related = related
+	v.Published, err = cves.ParseCVE5Timestamp(cve.Metadata.DatePublished)
+	if err != nil {
+		notes = append(notes, "Published date failed to parse, setting time to now")
+		v.Published = time.Now()
+	}
+	v.Modified, err = cves.ParseCVE5Timestamp(cve.Metadata.DateUpdated)
+	if err != nil {
+		notes = append(notes, "Modified date failed to parse, setting time to now")
+		v.Modified = time.Now()
+	}
+	v.References = vulns.ClassifyReferences(refs)
+	// Add affected version
+	// VERSIONS WILL BE ADDED IN ANOTHER PR
+	v.DatabaseSpecific = make(map[string]interface{})
+	CPEs := vulns.GetCPEs(cve.Containers.CNA.CPEApplicability)
+	if len(CPEs) != 0 {
+		v.DatabaseSpecific["CPE"] = vulns.Unique(CPEs)
+	}
+
+	// TODO: add CWEs
+
+	// Find metrics across CNA and adp
+	var metrics []cves.Metrics
+	if len(cve.Containers.CNA.Metrics) != 0 {
+		metrics = append(metrics, cve.Containers.CNA.Metrics...)
+	}
+	for _, adp := range cve.Containers.ADP {
+		if len(adp.Metrics) != 0 {
+			metrics = append(metrics, adp.Metrics...)
+		}
+	}
+	for _, m := range metrics {
+		v.AddSeverity(m)
+	}
+	return &v, notes
+}
```

Comment:
The comments in the functions is really nice! 

I would further refactor it along those commented sections.

Currently this function does quite a few different things as identified by the comments for each section.

You can group the "save OSV record to disk" section into it's own  function, and same for "save metrics to disk"

Then you will essentially have 4 function calls that are really clear:

- Make CVE record
- Save CVE record
- Make Metrics
- Save Metrics